product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api/1.1.2-create-rest-api-by-import-oas-document/src/test/java/org/wso2/am/scenario/tests/rest/api/creation/RESTApiCreationUsingOASDocTestCase.java

```diff
@@ -82,6 +86,11 @@ public void createApiWithValidOAS2DocumentAsJSONFile() throws Exception {
         HttpResponse getResponse = apiPublisher.getAPI(apiName, "admin", apiVersion);
         String resource = (new JSONObject(getResponse.getData())).getJSONObject("api").get("resources").toString();
         Assert.assertTrue(resource != "null", "API resource was not imported correctly");
+
+        //Assert resources
+        assertGETResource(((JSONArray)((JSONArray)(new JSONObject(getResponse.getData())).getJSONObject("api").
```

Comment:
why cast twice to JSONArray?

Reply:
get("templates") is also returns a jsonarray which we use to access each variable such as path.Hence the casting.

Reply:
I feel it's better to do those castings in two separate instructions for readability purpose. Otherwise, it's hard to interpret what is happening.

Reply:
+1. changed.